shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -103,10 +173,14 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
     [self.flutterViewController removeKeyResponder:self];
     [_textInputContext deactivate];
   } else if ([method isEqualToString:kClearClientMethod]) {
-    self.activeModel = nil;
+    _clientID = nil;
+    _inputAction = nil;
+    _inputType = nil;
+    _active_model = nullptr;
```

Comment:
Optional nit: maybe we should have a trivial wrapper class for all of these things that are bound up together, so they can be set/cleared as a unit instead of people needing to remember to keep these changes in sync.

Reply:
Was thinking about this. Specifically, I think text affinity could probably be migrated to TextInputModel since it's similar to other range/cursor pos/etc. data there. These fields might belong better in a separate text input context (or similar) class, as you say.

Will do this in a followup patch across all embedders.